### PR TITLE
Shrine images

### PR DESCRIPTION
**Pull request by GabrielLyonB**
## Objective
- Update `ImageBuilder` to allow the use of shrine images.

## Description
- I included [`shrine`](https://shrinerb.com/docs/getting-started) and [`image_processing`](https://github.com/janko/image_processing) as a developer dependency.
- I created a shrine initializer in the dummy app.
- I included the `picture_data` column in the `Invoice` modelo to try the shrine image uploader.
- I included in the `ImageBuilder` a case for shrine objects. This includes the use of derivatives.
- I tested the `ImageBuilder` based on the tests for paperclip.

### CODE REVIEW

File: spec/dummy/app/uploaders/image_uploader.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require 'image_processing/vips'
+
+class ImageUploader < Shrine
+  DERIVATIVE_SIZES = {
+    jpg_small: { size: [100, 100], type: 'jpg' }
+  }
+
+  Attacher.derivatives do |original|
+    vips = ImageProcessing::Vips.source(original)
```
Comment (gmq) [inline spec/dummy/app/uploaders/image_uploader.rb:9]: No usaría `vips` en la dummy app, es mucho más probable tener imagemagick instalado y paperclip ya lo usaba.

Comment (GabrielLyonB) [reply spec/dummy/app/uploaders/image_uploader.rb:9]: decidimos dejarlo tal cual después de discutirlo internamente

File: README.md [hunk 2]
```diff
@@ -78,13 +78,13 @@ Displays a paperclip link with attachment related icon into index and show views
 
 [Read more!](docs/paperclip_attachment.md)
 
-#### Paperclip Image
+#### Images
 
-Displays a paperclip image into index and show views
+Display images in the index and show views. This is implementation supports [Shrine](https://github.com/shrinerb/shrine) and [Paperclip](https://github.com/thoughtbot/paperclip).
```
Comment (gmq) [inline README.md:83]: `This implementation`

File: docs/images.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+<img src="./images/paperclip-image-column.png" height="400" />
+
+> You can pass `style` attribute matching paperclip's style definition
+
+## Shrine
+
+### Image Row
+
+The Shrine implementation receives an optional `image_options` which is then use as the options for the [`image_tag`](https://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag) method.
```
Comment (gmq) [inline docs/images.md:33]: `then used`

File: docs/images.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    image_row :photo, image_options: { width: 400 }
+  end
+end
+```
+
+<img src="./images/paperclip-image-row.png" height="400" />
+
+### Image Column
+
+To use the Shrine [derivatives](https://shrinerb.com/docs/plugins/derivatives) you could use the `style` option with the name of the derivative like the example below. In this case you would need to have a derivative created with the `'jpg_small'` name.
```
Comment (gmq) [inline docs/images.md:46]: Acá cambiaría `could` por `can`, ya que `could` en este caso implica que se puede hacer de otra manera.

**Review by gmq — APPROVED** [review]

**Pull request merged**